File: java/src/test/java/problems1_100/RemoveDuplicatesFromSortedListTest.java
package problems1_100;

import concepts.SinglyListNode;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class RemoveDuplicatesFromSortedListTest extends SampledTest {
    @Test
    public void test() {
        for (RemoveDuplicatesFromSortedList solution : RemoveDuplicatesFromSortedList.values()) {
            for (RemoveDuplicatesFromSortedListSample sample
                : getSamples(RemoveDuplicatesFromSortedListSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.deleteDuplicates(SinglyListNode.of(sample.input.head)))
                    .isEqualTo(SinglyListNode.of(sample.output));
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/RemoveDuplicatesFromSortedListTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import concepts.SinglyListNode
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import kotlin.test.Test

class RemoveDuplicatesFromSortedListTest : SampledTest() {
    @Test
    fun test() =
        RemoveDuplicatesFromSortedList.entries.forEach { solution ->
            getSamples(Array<RemoveDuplicatesFromSortedListSample>::class.java)
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(solution.deleteDuplicates(SinglyListNode.of(*input.head)))
                        .isEqualTo(SinglyListNode.of(*output))
                }
        }
}
